### PR TITLE
MenuRenderer: Guard against open/close handlers firing incorrectly

### PR DESCRIPTION
**Pull request by michaeltaranto**

### CODE REVIEW

File: lib/components/MenuRenderer/MenuRenderer.tsx [hunk 1]
```diff
@@ -166,16 +166,20 @@ export const MenuRenderer = ({
...
 
   useEffect(() => {
-    if (open) {
-      hasOpened.current = true;
+    if (lastOpen.current === open) {
+      return;
+    }
 
+    if (open) {
       if (typeof onOpen === 'function') {
```
Comment (askoufis) [inline lib/components/MenuRenderer/MenuRenderer.tsx:174]: Any reason not to collapse these two `if`s into one?

**Review by askoufis — APPROVED** [review]
Looks good, just one non-blocking nitpick.

File: lib/components/MenuRenderer/MenuRenderer.tsx (hunk 1 above)
Comment (michaeltaranto) [reply lib/components/MenuRenderer/MenuRenderer.tsx:174]: Yep, because it `onOpen` is not a function we don't want to trigger the else clause.

Comment (askoufis) [reply lib/components/MenuRenderer/MenuRenderer.tsx:174]: But wouldn't `open && typeof onOpen === 'function'` achieve that, or am I missing something?

**Pull request merged**